go/prometheusexporter/collector/collector.go

```diff
@@ -201,6 +243,18 @@ func labelsFromLabelPairs(pairs []*promclient.LabelPair) prometheus.Labels {
 	return labels
 }
 
+func (c *Collector) deleteSummaryLabels(summary *prometheus.SummaryVec, labels prometheus.Labels) bool {
+	c.mutex.Lock()
```

Comment:
I'm a bit unsure why we need to lock the mutex here. We are not accessing the stream and we are just changing the value of the summary itself. This shouldn't have an impact to the pointer maintained by the `c *Collector` or the registration of said summary in the registry.

Reply:
You might be right. In any case, the lock is in the wrong place. The pointer access to the histogramvec needs to be protected, but the code doesn't do this.